ui/src/ProcessDetails.js

```diff
@@ -0,0 +1,189 @@
+    }
+  }
+
+  componentDidMount() {
+    this.getJobs()
+
+    this.interval = setInterval(
+      () => this.getJobs(),
+      1000
+    )
+  }
+
+  handleUpload = (e) => {
+    // Display the progress modal dialog
+    this.setState({
+      uploadInProgress: true,
+    })
+
+    const formData = new FormData();
+    formData.append("file", e.target.files[0]);
+    formData.append("bulkProcess", this.props.bulkProcess['bulkProcess'])
+
+    // Send the file data to the backend
+    axios.request({
+      method: "post",
+      url: "/upload",
+      data: formData,
+      headers: {
+        'Content-Type': 'multipart/form-data'
+      },
+      onUploadProgress: (p) => {
+        console.log(p);
+
+        // Update file upload progress
+        this.setState({
+          fileProgress: p.loaded / p.total
+        })
+      }
+
+    }).then(data => {
+      // Hide the progress dialog and flash the snackbar message
+      this.setState({
+        fileProgress: 1.0,
+        fileUploadSuccess: true,
+        uploadInProgress: false,
+      })
+
+      this.getJobs()
+    })
+  }
+
+  handleClose = (event, reason) => {
+    // Ignore clickaways so that the dialog is modal
+    if (reason === 'clickaway') {
+      return;
+    }
+
+    this.setState({
+      fileUploadSuccess: false,
+    })
+  }
+
+  getJobs = async () => {
+    const response = await fetch('/job?bulkProcess=' + this.props.bulkProcess['bulkProcess'])
+    const jobs = await response.json()
+
+    this.setState({ jobs: jobs })
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.interval)
+  }
+
+  handleOpenDetails = (index) => {
+    this.setState({ showDetails: true, selectedJob: index })
+  }
+
+  handleClosedDetails = () => {
+    this.setState({ showDetails: false })
+  }
+
+  render() {
+    const jobTableRows = this.state.jobs.map((job, index) => (
+      <TableRow key={job.createdAt}>
+        <TableCell component="th" scope="row">
+          {job.fileName}
+        </TableCell>
+        <TableCell>{job.createdAt}</TableCell>
+        <TableCell align="right">
+          <Button
+            onClick={() => this.handleOpenDetails(index)}
+            variant="contained">
+            {job.jobStatus} {!job.jobStatus.startsWith('PROCESSED') && <CircularProgress size={15} style={{ marginLeft: 10 }} />}
+          </Button>
+        </TableCell>
+      </TableRow>
+    ))
+
+    return (
+      <Grid>
+        <Paper elevation={3} style={{ margin: 10, padding: 10 }}>
+          <Typography variant="inherit" color="inherit" style={{ margin: 10, padding: 10 }}>
+            Please upload a {this.props.bulkProcess['title']} bulk file for processing
+            </Typography>
+          <input
+            accept="csv/*"
+            style={{ display: 'none' }}
+            id="contained-button-file"
+            type="file"
+            onChange={(e) => {
+              this.handleUpload(e)
+            }}
```

Comment:
Seen something weird happen with the input. If you upload a file and then click the upload button and then cancel on the file select screen, it will get stuck on `Uploading file 100%` and I can see in the console theres a 400 bad request for the upload url.
<img width="1680" alt="Screenshot 2021-05-24 at 12 11 35" src="https://user-images.githubusercontent.com/35301276/119339603-450db600-bc89-11eb-8603-3ece81c5c3e9.png">

Reply:
think something like `e.target.value = null;` could fix it inside of `handleUpload` once the upload is done